### DISCUSSION TITLE
Refactor File interface

### DISCUSSION THREAD

**Discussion by KaiVolland** [terrestris/shogun#215] [Ideas]
Currently there is no way to access a single file entity via REST. You can only get the file itself.

The default way in all other interfaces is that we get a JSON repsonse if we send a GET request.

e.g.:
```http
http://localhost:8080/shogun-boot/imagefiles/44
```
IMHO this should return:
```js
{
  id: 44,
  created: "2020-12-23T09:50:33.397+00:00",
  modified: "2020-12-23T09:50:33.397+00:00",
  fileUuid: "df0299b0-8827-4ab8-9346-2785ba004573",
  active: true,
  fileName: "lord_forntelroy.jpg",
  fileType: "image/jpeg",
  width: 654,
  height: 800
}
```
... and if we can't distinguish between `id` and `uuid` in the Requestparam, the actual file/image should be offered on a different endpoint. Like we already do with the thumbnail.

**Comment by dnlkoch** [comment]
In terms of REST I would prefer a separate subpath for getting the (image) file contents. Proposal (basically as you already suggested):

#### Getting the entity info:

```
http://localhost:8080/shogun-boot/files/{UUID}
```

⤋                 

```
{
  id: 44,
  fileUuid: "{UUID}"
  (…)
}
```

#### Getting the file content:

```
http://localhost:8080/shogun-boot/files/{UUID}/get
```

⤋                 

```
( ❛ ͜ʖ ❛ )
```

**Reply by KaiVolland** [reply]
`get` could be more precise like `getFile`

**Reply by dnlkoch** [reply]
In order to be more in sync with the existing `thumbnail` endpoint, I think we can skip the `get` part as well. It should be clear as it's mapped to a `GET` anyway. How about simply `file`?
